packages/react-hooks/src/ConnectionProvider.tsx

```diff
@@ -15,14 +15,71 @@ import {
   addRecord,
 } from './recordUtils'
 
+export type useConnectionsOptions = {
+  excludedTypes?: [ConnectionType | string]
+  connectionState?: DidExchangeState
+}
+
 const ConnectionContext = createContext<RecordsState<ConnectionRecord> | undefined>(undefined)
 
-export const useConnections = () => {
+/**
+ * This method retreives the connection context for the current agent.
+ * From this you can access all connection records for the agent.
+ * @param options options for useConnections hook, lets us filter out specific types and limit states
+ * @returns a connection context containing information about the agents connections
+ */
+export const useConnections = (options: useConnectionsOptions = {}) => {
   const connectionContext = useContext(ConnectionContext)
   if (!connectionContext) {
     throw new Error('useConnections must be used within a ConnectionContextProvider')
   }
-  return connectionContext
+
+  let connections = connectionContext.records
+
+  // Only run this code if we don't have options.excludedTypes to limit ammount of loops
+  if (options.connectionState && !options.excludedTypes) {
+    const filteredByState = useMemo(
+      () => connections.filter((c: ConnectionRecord) => c.state === options.connectionState),
+      [connections, options.connectionState]
+    )
+    connections = filteredByState
+  }
+
+  if (options.excludedTypes) {
+    const filteredByType = useMemo(
+      () =>
+        connections.filter((record: ConnectionRecord) => {
+          const recordTypes = (record.getTag('connectionType') as [string] | null) || []
+          // Filter out by state if needed
+          if (options.connectionState && options.connectionState !== record.state) {
+            return false
+          }
+          for (const type of recordTypes) {
+            if (options.excludedTypes?.includes(type)) return false
+          }
+          return true
+        }),
+      [connections, options.excludedTypes]
+    )
+    connections = filteredByType
+  }
```

Comment:
I think we can simplify this by creating two filters that filter based on their own criteria. This way the excludedTypes filter doesn't need to know about the state filter and vice versa.

Reply:
We were trying to give users the ability to filter by both connection type and connection state at the same time with a single hook as we didn't see a clean way to do that if they are separate hooks. It should only filter by both if the options indicate to do so, and will default to doing neither.